week1/index.js

```diff
@@ -1,25 +1,67 @@
-// To run this file: node index.js
-// Then in your browser: http://localhost:8080
-var http = require('http');
-
-var port = 8080;
-
-var server = http.createServer();
-
-// Start the HTTP server, start listening for requests
-server.listen(port, function(error) {
-  if (error) {
-    console.log(error);
-  } else {
-    console.log('api listening on port', port);
-  }
+/*global console*/
+"use strict";
+//here the global variables
+let state = 10;
+const port = 3000;
+// Start the HTTP server
+let  http = require('http');
+let  server = http.createServer();
+//function to build the response and specifies the data type
+function updateResponse(response, message) {
+    response.setHeader('content-type', 'text/html');
+    response.write('<html><head></head><body><h1>' + message + '</h1></body></html>');
+    response.end();
+}
+//The function will show the error if the user wrote a url that does not exist, this Function Will be called later
+function updateError(error, response) {
+	response.setHeader('content-type', 'text/html');
+	response.write('<html><head></head><body><h1>' + error + '</h1></body></html>');
+	response.end();
+}
+//this Function within the event to check for URL and The state changes accordingly, as does the appropriate message
+server.on('request', function makeRequest(request, response) {
+	var message;
+	switch (request.url) {
+		case '/':
+		//localhost:3000
+			message = 'You got it';
+			updateResponse(response, message);
+			break;
+		case '/state':
+		//localhost:3000/state
```

Comment:
Awesome use of comments to explain exactly in what situation this particular codepath gets executed.